server/controllers/items/lib/bulk_update.js

```diff
@@ -0,0 +1,45 @@
+const items_ = require('controllers/items/lib/items')
+const Item = require('models/item')
+const { emit } = require('lib/radio')
+const { partition } = require('lodash')
+const _ = require('builders/utils')
+const { validateVisibilityKeys } = require('lib/visibility/visibility')
+const { validateShelves } = require('controllers/items/lib/validate_item_async')
+
+const bulkItemsUpdate = async ({ reqUserId, ids, attribute, value, attempt = 0, previousUpdates = [] }) => {
+  const itemUpdateData = { [attribute]: value }
+  const currentItems = await items_.byIds(ids)
+  const formattedItems = currentItems.map(currentItem => Item.update(reqUserId, itemUpdateData, currentItem))
+  await validateValue({ attribute, value, reqUserId })
+  try {
+    const successfulUpdates = await items_.db.bulk(formattedItems)
+    await emit('user:inventory:update', reqUserId)
+    return previousUpdates.concat(successfulUpdates)
+  } catch (err) {
+    if (attempt > 10) throw err
+    const { body } = err.context
+    const [ failedUpdates, successfulUpdates ] = partition(body, hasError)
```

Comment:
[non-blocking] having to refer to line 4 to figure out that `partition` is a lodash function sucks. I preferred the previous inline notation `_.partition`. I believe we should be driven by our own readability, rather than what IDEs forces us to do.

Reply:
See the other comment. I would be in favor of installing `@types/lodash` so that hovering the function gives you more info about what it is and its interface

Reply:
relying on third party services, such as certain proprietary IDE features, is a dependency that makes it less stable than  plain old declaration in the beginning of the file.

Reply:
what proprietary feature?
- `@types/lodash` is MIT licensed
- vscodium and it's parent project are also MIT licensed
- typescript, which I assume is the language used for those type declarations, uses Apache License 2.0

Reply:
=> proprietary "program" feature.
my point being, the more IDE agnostic the code is, the more stable. Its a dependency like another. 

Back on how to import this kind of external dependency functions, when i just see `partition` in the code, i would like to not have to refer to the dependency documentation and use a markup system like the current `_.foo()` to not have to think or hover or do whatever to find out what's this function anyway.

PS: note that i avoid trolling on who owns VSC & TS, since licence can change when they feel like it